crates/services/p2p/src/peer_manager.rs

```diff
@@ -553,115 +86,127 @@ impl PeerManager {
         }
     }
 
-    fn total_peers_connected(&self) -> usize {
+    pub fn is_reserved_peer(&self, peer_id: &PeerId) -> bool {
+        self.reserved_peers.contains(peer_id)
+    }
+
+    pub fn handle_peer_info_updated(
+        &mut self,
+        peer_id: &PeerId,
+        block_height: BlockHeight,
+    ) {
+        if let Some(previous_heartbeat) = self
+            .get_peer_info(peer_id)
+            .and_then(|info| info.heartbeat_data.seconds_since_last_heartbeat())
+        {
+            debug!(target: "fuel-p2p", "Previous hearbeat happened {:?} seconds ago", previous_heartbeat);
+        }
+
+        let heartbeat_data = HeartbeatData::new(block_height);
+
+        self.insert_peer_info(peer_id, PeerInfoInsert::HeartbeatData(heartbeat_data));
+    }
+
+    /// Returns `true` signaling that the peer should be disconnected
+    pub fn handle_peer_connected(
+        &mut self,
+        peer_id: &PeerId,
+        addresses: Vec<Multiaddr>,
+        initial_connection: bool,
+    ) -> bool {
+        if initial_connection {
+            self.handle_initial_connection(peer_id, addresses)
+        } else {
+            self.insert_peer_info(peer_id, PeerInfoInsert::Addresses(addresses));
+            false
+        }
+    }
+
+    pub fn handle_peer_identified(
+        &mut self,
+        peer_id: &PeerId,
+        addresses: Vec<Multiaddr>,
+        agent_version: String,
+    ) {
+        self.insert_peer_info(peer_id, PeerInfoInsert::ClientVersion(agent_version));
+        self.insert_peer_info(peer_id, PeerInfoInsert::Addresses(addresses));
+    }
+
+    pub fn batch_update_score_with_decay(&mut self) {
+        for peer_info in self.non_reserved_connected_peers.values_mut() {
+            peer_info.score *= DECAY_APP_SCORE;
+        }
+    }
+
+    pub fn update_peer_score_with(
+        &mut self,
+        peer_id: PeerId,
+        score: AppScore,
+    ) -> Option<PeerScoreUpdated> {
+        if let Some(peer) = self.non_reserved_connected_peers.get_mut(&peer_id) {
+            // score should not go over `MAX_PEER_SCORE`
+            let new_score = self.peer_score_config.max_score.min(peer.score + score);
+            peer.score = new_score;
+
+            let should_ban = new_score < self.peer_score_config.min_score_allowed;
+
+            Some(PeerScoreUpdated {
+                should_ban,
+                score: new_score,
+            })
+        } else {
+            log_missing_peer(&peer_id);
```

Comment:
This log message might be confusing if the peer is connected as a reserved peer.

Reply:
If you meant specifically in the context of `update_peer_score_with()`, that method works only with `non_reserved_connected_peers` HashMap so no reserved peer will be logged from here.

If you meant `log_missing_peer` in general, the (reserved) peer could still not be connected at a given point when an action was tried, so it logs that the peer is missing/disconnected.